src/include/migraphx/op/reduce_op.hpp

```diff
@@ -107,16 +108,40 @@ struct reduce_op : op_name<Derived>
 
     shape normalize_compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs, *this}.has(1);
-        auto s          = inputs.at(0);
-        auto lens       = s.lens();
-        auto tuned_axes = tune_axes(lens.size());
-        for(auto axis : tuned_axes)
+        check_shapes{inputs, *this, true}.has(1);
+        auto s = inputs.at(0);
+        if(s.dynamic())
         {
-            lens[axis] = 1;
+            std::vector<shape::dynamic_dimension> output_dyn_dims = {};
+
+            // create a dynamic dimensions vector that leaves out any axis named in this->axes.
+            for(size_t index = 0; index < s.dyn_dims().size(); index++)
+            {
+                auto name_it = std::find_if(this->axes.begin(), this->axes.end(), [&](auto& axis) {
```

Comment:
This should use the `tuned_axes` Otherwise negative axis wont work.

Reply:
It does work, even if accidentally.  A test of a negative axis is at https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/dynamic_reduce/test/op_shape_test.cpp#L1403